packages/click-to-react-component/src/ClickToComponent.js

```diff
@@ -44,12 +44,16 @@ export function ClickToComponent({ editor = 'vscode' }) {
         const url = `${editor}://file/${path}`
 
         event.preventDefault()
-        window.open(url)
+        if (onOpenFile) {
```

Comment:
Wow! Thanks for taking the initiative!  This definitely seems like a need.


What do you think about moving it up one level before the `path` is crafted.  Like, something that accepts `{ editor, source }`?

Such as `onOpenComponent({ source })` or similar?

I learned that different editors have different URL schemes, so it'd give the opportunity for people to customize it.

Then, our default implementation would look like:

```js
onOpenComponent = ({ editor, source }) => {
  const path = getPathToSource(source)
  const url = `${editor}://file/${path}`

  window.open(url);
}
```

Reply:
Oh I see, `source` has a file name, line & column. So e.g. for `vim` it would be `+{line} {file}`. The only drawback here is the user would have to manually import and call `getPathToSource` or write their own.

Is there a reason you have `editor` in `onOpenComponent: ({ editor, source }) => void`? This prop is passed by the same callsite that sets the callback, so they should already know the `editor`.

Reply:
No, no real reason.  You're right – no sense adding it to an API unless it actually makes sense.

Based off of #26 , I think these are actually two distinct needs.  So you can ignore my Qs :)

We'll probably end up with something like:

```js
<ClickToComponent
  getURLFromSource={customMapper}
  onOpenURL={customOpener}
/>
```